### PR TITLE
Adjusting logic to match intention

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Bug Fix

## Description

Unearthed in a deep read of the code during the review of forem/forem#16997

Please read from a "does this logic make sense" perspective.

## Related Tickets & Documents

forem/forem#16997

## QA Instructions, Screenshots, Recordings

None

### UI accessibility concerns?

None.

### CODE REVIEW

**Review by msarit — APPROVED** [review]
Honestly, the previous code did NOT make sense to me! 😅 The updated code does. 👌🏾

**Comment by jeremyf** [comment]
> Honestly, the previous code did NOT make sense to me! 😅 The updated code does. 👌🏾

The previous code had lived for 2.5 years untouched

File: app/controllers/dashboards_controller.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def show
     fetch_and_authorize_user
     target = @user
     # NOTE: This is a subtle policy check happening here that we are not encapsulating
-    not_authorized if params[:org_id] && !@user.org_admin?(params[:org_id] || @user.any_admin?)
```
Comment (djuber) [inline app/controllers/dashboards_controller.rb:24]: how did this ever work before?
`user.org_admin?(params[:org_id] || @user.any_admin?)`

File: app/controllers/dashboards_controller.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ def show
...
+    not_authorized if params[:org_id] && !(@user.org_admin?(params[:org_id]) || @user.any_admin?)
```
Comment (djuber) [inline app/controllers/dashboards_controller.rb:24]: when there was an org id param provided,
and neither of these are true
 - user is an org admin of this org
 - user is an admin 
 
then they shouldn't be viewing the organization's dashboard.

**Review by djuber — APPROVED** [review]

**Pull request merged**